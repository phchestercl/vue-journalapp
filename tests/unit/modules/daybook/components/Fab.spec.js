import { shallowMount } from '@vue/test-utils'
import Fab from '../../../../../src/modules/daybook/components/Fab.vue'
describe('Fab Componente',()=>{
    test('Snapshot',()=>{
        const wrapper = shallowMount(Fab)
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Utiliza clase por defecto',()=>{
        const wrapper =shallowMount(Fab)
        const clsDefault= wrapper.find('i').classes('fa-plus')
        expect(clsDefault).toBeTruthy
    })
    test('Enviar clase distinta a la clase por defecto',()=>{
        const wrapper = shallowMount(Fab,{
            props:{
                icon:'fa-circle'
            }
        })
        const cls=wrapper.find('i').classes('fa-circle')
        expect(cls).toBeTruthy()
    })
    test('Emite on:click al hacer click', async()=>{
        /* const mockEmit={
            $emit:jest.fn()
        } */
        const wrapper =shallowMount(Fab)
        wrapper.find('button').trigger('click')
        //console.log(wrapper)
        console.log(wrapper.emitted('on:click'))
        
        expect(wrapper.emitted('on:click')).toHaveLength(1)
    })
})