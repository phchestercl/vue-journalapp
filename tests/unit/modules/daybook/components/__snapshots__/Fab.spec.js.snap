// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Componente Snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
